Realm/Realm/DatabaseTypes/Accessors/IRealmAccessor.cs

```diff
@@ -0,0 +1,69 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Realms.Schema;
+
+namespace Realms
+{
+    public interface IRealmAccessor
+    {
+        bool IsManaged { get; }
+
+        bool IsValid { get; }
+
+        bool IsFrozen { get; }
+
+        Realm Realm { get; }
+
+        ObjectSchema ObjectSchema { get; }
+
+        int? HashCode { get; }
```

Comment:
I'm not sure I understood what you mean with this.

Reply:
Yeah - if the accessor overrides `GetHashCode`, we can use that.

Reply:
Makes sense, but this means that for unmanaged objects we're getting the hashcode of the accessor, instead of that os the object iself. 
I suppose that this is not an issue though